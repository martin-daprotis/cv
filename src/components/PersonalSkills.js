import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Typography,
  Box,
  Grid,
} from "@material-ui/core";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import EN from "../lang_source/EN.json";
import SP from "../lang_source/SP.json";
import { v4 as uuidv4 } from "uuid";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: "33.33%",
    flexShrink: 0,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
    float: "right",
  },
  position: {
    color: theme.palette.text.secondary,
  },
}));

export default function PersonalSkills({ language, t, sp = SP, en = EN }) {
  const classes = useStyles();
  const lang = language === "SP" ? sp : en;

  function Item({ panel, title, description }) {
    return (
      <Accordion
        TransitionProps={{
          timeout: 600,
        }}
      >
        <AccordionSummary
          expandIcon={<ExpandMoreIcon />}
          aria-controls="panelbh-content"
          id="panelbh-header"
        >
          <Grid
            container
            direction="row"
            justify="flex-start"
            alignItems="center"
          >
            <Typography variant="h6" color="textSecondary">
              {title}
            </Typography>
          </Grid>
        </AccordionSummary>
        <AccordionDetails>
          <Typography variant="subtitle1">{description}</Typography>
        </AccordionDetails>
      </Accordion>
    );
  }

  return (
    <div className={classes.root}>
      <Box
        display="flex"
        justifyContent="center"
        m={1}
        p={1}
        fontWeight="fontWeightBold"
      >
        <Typography variant="h4">{t("navbar.personal_skills")}</Typography>
      </Box>
      {lang.personal_skills.map((k, i) => {
        return (
          <Item
            key={uuidv4()}
            title={k.skill}
            description={k.description}
            panel={i}
          />
        );
      })}
    </div>
  );
}
