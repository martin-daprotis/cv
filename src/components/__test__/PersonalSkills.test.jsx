import React from 'react'
import { render, fireEvent, cleanup, waitFor, getByTestId ,logDOM} from '@testing-library/react'
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import {
  translate,
} from "react-switch-lang";
import PersonalSkills from '../PersonalSkills';


const mockSkillEN = {
  "personal_skills" :[
    {
      "skill":"Skill",
      "description":"Skill Description"
    },
  ]  
};
const mockSkillSP = {
  "personal_skills" :[
    {
      "skill":"Habilidad",
      "description":"Descripcion de la habilidad"
    },
  ]  
};

afterEach(cleanup)

describe('Spanish language specifying mocks', () => {
  it('render properly ',  () => {
    const RenderHOC = translate(({t}) => <PersonalSkills sp={mockSkillSP} en={mockSkillEN} t={t} language='SP'/>);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText("Habilidad")).toBeInTheDocument();
    expect(getByText(/descripcion de la habilidad/i)).toBeInTheDocument();
  })
})

describe('English language specifying mocks', () => {
  it('render properly ',  () => {
    const RenderHOC = translate(({t}) => <PersonalSkills sp={mockSkillSP} en={mockSkillEN} t={t} language='EN'/>);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText("Skill")).toBeInTheDocument();
    expect(getByText(/skill description/i)).toBeInTheDocument();
  })
})

describe('Spanish language default languages', () => {
  it('render properly ',  () => {
    const RenderHOC = translate(({t}) => <PersonalSkills t={t} language='SP'/>);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText("Adaptabilidad")).toBeInTheDocument();
    expect(getByText(/sumamente necesaria/i)).toBeInTheDocument();
  })
})

describe('English language default languages', () => {
  it('render properly ',  () => {
    const RenderHOC = translate(({t}) => <PersonalSkills t={t} language='EN'/>);
    const {getByText} = render(<RenderHOC/>)
    expect(getByText("Adaptability")).toBeInTheDocument();
    expect(getByText(/is mandatory/i)).toBeInTheDocument();
  })
})

it('matches snapshot',  () => {
  const RenderHOC = translate(({t}) => <PersonalSkills t={t} language='EN'/>);
const wrapper = render(<RenderHOC/>)
expect(wrapper.asFragment()).toMatchSnapshot()
})