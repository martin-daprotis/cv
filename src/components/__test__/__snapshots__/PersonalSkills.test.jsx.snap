// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-21"
  >
    <div
      class="MuiBox-root MuiBox-root-25"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        navbar.personal_skills
      </h4>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panelbh-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panelbh-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary"
            >
              Communication
            </h6>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panelbh-header"
              id="panelbh-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  During my work experience as a Full Stack Web Developer I had to meet with the client in order to gather requirements to start the development process of new Systems. Additionally, I was responsible for answering client enquiries by phone solving their problems and giving support on systems which I was maintaining.
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panelbh-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panelbh-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary"
            >
              Responsibility
            </h6>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panelbh-header"
              id="panelbh-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  I was responsible for developing and maintaining systems with sensible data such as employee salaries or bank transactions systems. These systems were crucial to make the monthly payments of the 4000+ employees of the company.
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panelbh-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panelbh-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary"
            >
              Teamwork
            </h6>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panelbh-header"
              id="panelbh-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  Even though I had to work in some cases on my own it is always better to face challenges as a team. For more than 5 year now, I have been part of a development team, in which all members take part in a collaborative way with SCRUM methodologies.
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panelbh-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panelbh-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextSecondary"
            >
              Adaptability
            </h6>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panelbh-header"
              id="panelbh-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1"
                >
                  This skill is mandatory to work in web development, often when I encounter new challenges it is necessary to investigate and start learning new technologies, packages or libraries, and apply if suitable. To be able to achieve this it is important to adapt to constant changes in technology.
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
